src/NHibernate/Persister/Collection/ICollectionPersister.cs

```diff
@@ -283,4 +284,21 @@ public partial interface ICollectionPersister
 		/// </summary>
 		object NotFoundObject { get; }
 	}
+
+	public static class CollectionPersisterExtensions
+	{
+		/// <summary>
+		/// Get the batch size of a collection persister.
+		/// </summary>
+		//6.0 TODO: Merge into ICollectionPersister and convert to a property.
+		public static int GetBatchSize(this ICollectionPersister persister)
+		{
+			if (persister is AbstractCollectionPersister acp)
+			{
+				return acp.GetBatchSize();
+			}
+
+			throw new InvalidOperationException("Only persisters of AbstractCollectionPersister are supported.");
```

Comment:
@fredericDelaporte I think returning 1 here would fix this possible breaking change, wouldn't it?